api/client/lib.py

```diff
@@ -349,12 +394,77 @@ def rank_series_by_source(access_token, api_host, series_list):
+
+    >>> format_list_of_series([{
+    ...     'series': { 'metricId': 1, 'itemId': 2, 'regionId': 3, 'belongsTo': { 'itemId': 22 } },
+    ...     'data': [
+    ...         ['2001-01-01', '2001-12-31', 123]
+    ...     ]
+    ... }]) == [
+    ...   { 'start_date': '2001-01-01',
+    ...     'end_date': '2001-12-31',
+    ...     'value': 123,
+    ...     'unit_id': None,
+    ...     'reporting_date': None,
+    ...     'metric_id': 1,
+    ...     'item_id': 2,
+    ...     'region_id': 3,
+    ...     'partner_region_id': 0,
+    ...     'frequency_id': None,
+    ...     'source_id': None,
+    ...     'belongs_to': { 'item_id': 22 }
+    ... } ]
+    True
+
+    """
+    if not isinstance(series_list, list):
+        # If the output is an error or None or something else that's not a list, just propagate
+        return series_list
+    output = []
+    for series in series_list:
+        if not (isinstance(series, dict) and isinstance(series.get('data', []), list)):
+            continue
+        # All the belongsTo keys are in camelCase. Convert them to snake_case.
+        # Only need to do this once per series, so do this outside of the list
+        # comprehension and save to a variable to avoid duplicate work:
+        belongs_to = camel_to_snake_dict(series.get('series', {}).get('belongsTo', {}))
+        output += [{
+            'start_date': point[0],
+            'end_date': point[1],
+            'value': point[2],
+            # list_of_series has unit_id in the series attributes currently. Does
+            # not allow for mixed units in the same series
+            'unit_id': series['series'].get('unitId', None),
+            # If a point does not have reporting_date, use None
+            'reporting_date': point[3] if len(point) > 3 else None,
+            # Series attributes:
+            'metric_id': series['series'].get('metricId', None),
+            'item_id': series['series'].get('itemId', None),
+            'region_id': series['series'].get('regionId', None),
+            'partner_region_id': series['series'].get('partnerRegionId', 0),
+            'frequency_id': series['series'].get('frequencyId', None),
+            'source_id': series['series'].get('sourceId', None),
+            # belongs_to is consistent with the series the user requested. So if an
+            # expansion happened on the server side, the user can reconstruct what
+            # results came from which request.
+            'belongs_to': belongs_to
+        } for point in series.get('data', [])]
+    return output
+
+
 def get_data_points(access_token, api_host, **selection):
+    assert 'metric_id' in selection, 'metric_id is required'
+    assert 'item_id' in selection, 'item_id is required'
+    assert 'region_id' in selection, 'region_id is required'
+    assert 'source_id' in selection, 'source_id is required'
+    assert 'frequency_id' in selection, 'frequency_id is required'
```

Comment:
This is not-backward compatible. currently get_data_points allows some of these to be unspecified which can be useful for advanced users.  Let's discuss offline.

Reply:
Removed for now. Not strictly necessary for this feature